src/presentation/controllers/UserController.ts

```diff
@@ -1,12 +1,17 @@
 import { Request, Response } from 'express'
-import UserService from '../../domain/user/services/UserService'
+import { inject, injectable } from 'tsyringe'
+import IUserController from '../../interfaces/IUserController'
+import IUserService from '../../interfaces/IUserService'
 
-class UserController {
- static async handle(req: Request, res: Response) {
-  const criaUser = await UserService.criaUser(req.body)
+@injectable()
+export default class UserController implements IUserController {
+  userService: IUserService
+  constructor (@inject('UserService') userService: IUserService) {
+    this.userService = userService
+  }
 
-  res.status(criaUser.code).json(criaUser.msg)
- }
+  handle = async (req: Request, res: Response): Promise<void> => {
+    const criaUser = await this.userService.criaUser(req.body)
+    res.status(criaUser.code).json(criaUser.msg)
```

Comment:
Criação sempre será 201(Preferencialmente) ou 200, não faz sentido retornar do código
```suggestion
    const result = await this.userService.criaUser(req.body)
    res.status(201).json(result.msg)
```

Reply:
Resolvido